### DISCUSSION TITLE
Disable stacking git icons on folders

### DISCUSSION THREAD

**Discussion by Machine-Maker** [nvim-tree/nvim-tree.lua#2192] [Q&A]
I'd like to limit the number of git icons that are appened (or prepended) to folder names in the tree view. I haven't tried to engineer a situation where all of them are there, but 4 can easily be present which I feel clutters up the tree view. 

My dream, is for there to be complete configuration of how git statuses map to specific icon, and configurable grouping to have 1 icon cover multiple statuses, but that's kinda more than I'd like to do now. Just looking for a way to limit the number of icons for folders.

**Comment by alex-courtis** [comment]
There may only be two git status icons: staged and unstaged. When using `renderer.icons.git_placement` `signcolumn`, only the first will be shown.

See combinations: https://github.com/nvim-tree/nvim-tree.lua/blob/1830e5e8a46b32575130bf04bbe0a773999ccb73/lua/nvim-tree/renderer/components/git.lua#L104

Please try using `signcolumn` placement as an experiment to see whether first-only meets your needs.

**Comment by Machine-Maker** [comment]
Yeah, I tried that, but I do want the icon after the folder name. 
~~Also I think you can get 3?~~
actually 4
<img width="230" alt="image" src="https://user-images.githubusercontent.com/15055071/236964522-42cdd877-b78c-417c-80c5-5202c7b0e7f8.png">

**Reply by alex-courtis** [reply]
That's expected.

I'm interested in whether the priority in the sign column meets your needs.

If it does, we can use that same priority for a single icon afterwards.

**Reply by Machine-Maker** [reply]
The sign column makes the icons appear, well, in the sign column at the far left side. I'd like the icons to appear after the name of the folder

**Reply by alex-courtis** [reply]
> The sign column makes the icons appear, well, in the sign column at the far left side. I'd like the icons to appear after the name of the folder

Are you happy with the priority of the single sign in the column?

That question shapes a possible solution.

**Reply by Machine-Maker** [reply]
Oh yeah, that works then.

**Reply by alex-courtis** [reply]
#2229 2229
